src/pip/_internal/format_control.py

```diff
@@ -0,0 +1,64 @@
+from pip._vendor.packaging.utils import canonicalize_name
+
+
+class FormatControl(object):
+    """This class has two fields, no_binary and only_binary.
```

Comment:
Perhaps this could use a better description:

```
A helper class for controlling formats from which packages are installed.

[and some lines on how it should be used?]
```